srtcore/threadname.h

```diff
@@ -16,19 +16,15 @@ written by
 #ifndef INC_SRT_THREADNAME_H
 #define INC_SRT_THREADNAME_H
 
-#ifdef __linux__
+#if defined(__linux__)
 
 #include <sys/prctl.h>
 #include <cstdio>
 
 class ThreadName
 {
-    char old_name[128];
-    char new_name[128];
-    bool good;
-
 public:
-    static const size_t BUFSIZE = 128;
+    static const size_t BUFSIZE = 16;
```

Comment:
Please add a comment with motivation for 16 bytes buffer.

```suggestion
    // PR_SET_NAME and PR_GET_NAME (since Linux 2.6.9)
    // both require the name buffer to be up to 16 bytes long, including the terminating null byte.
    // (If the length of the string, including the terminating null byte, exceeds 16 bytes,
    // the string is silently truncated.)
    static const size_t BUFSIZE = 16;
```

Reply:
> Please add a comment with motivation for 16 bytes buffer.

I have given the reason in the commit message:
1. It's documented in the man page. Use a larger BUFSIZE is misleading.
2. The thread name would be truncated by PR_SET_NAME, so we don't do
   the truncation.

Yes it's a good idea to include the reference in the code. I will update the patch.